src/components/DocExplorer.js

```diff
@@ -43,23 +43,31 @@ export class DocExplorer extends React.Component {
   }
 
   // Public API
-  showDoc(typeOrField) {
-    var navStack = this.state.navStack;
-    var isCurrentlyShown =
-      navStack.length > 0 && navStack[navStack.length - 1] === typeOrField;
-    if (!isCurrentlyShown) {
-      navStack = navStack.concat([ typeOrField ]);
+  showDoc(navItem) {
```

Comment:
This API gets less clear as a result of this change, perhaps there's a better way to do this. Search should probably enter the nav stack via a different API to preserve this